meta-facebook/yv35-hd/src/ipmi/plat_ipmi.c

```diff
@@ -134,3 +135,131 @@ void OEM_1S_GET_BIOS_VERSION(ipmi_msg *msg)
 	msg->completion_code = CC_SUCCESS;
 	return;
 }
+
+#ifdef CONFIG_I2C_IPMB_SLAVE
+void OEM_1S_MSG_OUT(ipmi_msg *msg)
+{
+	CHECK_NULL_ARG(msg);
+#if MAX_IPMB_IDX
+	uint8_t target_IF;
+	ipmb_error status;
+	ipmi_msg *bridge_msg = NULL;
+
+	// If command is valid the default cc is CC_INVALID_CMD, set cc to CC_SUCCESS before we execute the command.
+	// If the default cc is CC_INVALID_IANA, call ipmb_send_response for this invalid command.
+	if (msg->completion_code != CC_INVALID_IANA) {
+		msg->completion_code = CC_SUCCESS;
+	}
+
+	// Should input target, netfn, cmd
+	if (msg->data_len <= 2) {
+		msg->completion_code = CC_INVALID_LENGTH;
+	}
+
+	CARD_STATUS _1ou_status = get_1ou_status();
+	target_IF = msg->data[0];
+
+	switch (target_IF) {
+	case PEER_BMC_IPMB:
+		switch (msg->InF_source) {
+		case SLOT1_BIC:
+			target_IF = msg->data[0] = SLOT3_BIC;
+			break;
+		case SLOT3_BIC:
+			target_IF = msg->data[0] = SLOT1_BIC;
+			break;
+		default:
+			msg->completion_code = CC_INVALID_DATA_FIELD;
+			break;
+		}
+
+		if (msg->data[1] == NETFN_STORAGE_REQ) {
+			msg->data[1] = msg->data[1] << 2;
+		}
+		break;
+	case EXP2_IPMB:
+		if (_1ou_status.card_type == TYPE_1OU_EXP_WITH_E1S) {
+			target_IF = EXP1_IPMB;
+		}
+		break;
+	case EXP4_IPMB:
+		target_IF = EXP3_IPMB;
+		break;
+	default:
+		break;
+	}
+
+	// Bridge to invalid or disabled interface
+	if ((IPMB_inf_index_map[target_IF] == RESERVED) ||
+	    (IPMB_config_table[IPMB_inf_index_map[target_IF]].interface == RESERVED_IF) ||
+	    (IPMB_config_table[IPMB_inf_index_map[target_IF]].enable_status == DISABLE)) {
+		LOG_ERR("OEM_MSG_OUT: Invalid bridge interface: %x", target_IF);
+		msg->completion_code = CC_NOT_SUPP_IN_CURR_STATE;
+	}
+
+	// only send to target while msg is valid
+	if (msg->completion_code == CC_SUCCESS) {
+		bridge_msg = (ipmi_msg *)malloc(sizeof(ipmi_msg));
+		if (bridge_msg == NULL) {
+			msg->completion_code = CC_OUT_OF_SPACE;
```

Comment:
It would be safer to exit here instead of allowing fall-through to the end of the function.  Early return is preferred as any new code added to the end of the function later would lead to unintended execution in a failed state.

Reply:
Modify as recommendation